kernel/src/main/scala/org/typelevel/sbt/kernel/V.scala

```diff
@@ -51,8 +51,8 @@ private[sbt] final case class V(
         if (z != 0) return z
         (this.prerelease, that.prerelease) match {
           case (None, None) => 0
-          case (Some(_), None) => 1
-          case (None, Some(_)) => -1
+          case (Some(_), None) => -1
+          case (None, Some(_)) => 1
           case (Some(thisPrerelease), Some(thatPrerelease)) =>
             // TODO not great, but not everyone uses Ms and RCs
```

Comment:
This has a ton of corner cases.  Which corners (e.g., where to look for numbers, "well-known qualifiers") depends on which spec (e.g., SemVer, Coursier, Ivy, Maven).  But this PR will make it do the right thing a significantly higher percentage of the time in practice in all of them.